File: python/disease.py
'''
Author: kvaden
10 April 2020
v1.0
'''

class Disease:
    def __init__(self, color=None, cured=False, eradicated=False):
        self.color = color
        self.cured = cured
        self.eradicated = eradicated

    def __str__(self):
        return 'The {} disease. Cured: {}. Eradicated: {}.'.format(self.color, self.cured, self.eradicated)

File: models/Disease.js
const { Model, DataTypes } = require('sequelize');
const sequelize = require('../config/connection');

class Disease extends Model {}

Disease.init(
  {
    id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      primaryKey: true,
      autoIncrement: true,
    },
    color: {
      type: DataTypes.STRING
    },
    cubes: {
      type: DataTypes.INTEGER
    },
    cured: {
      type: DataTypes.BOOLEAN
    },
    eradicated: {
        type: DataTypes.BOOLEAN
    }
  },
  {
    sequelize,
    timestamps: false,
    freezeTableName: true,
    underscored: true,
    modelName: 'disease'
  }
);

module.exports = Disease;